ui/lib/kv/addon/components/page/list.hbs

```diff
@@ -3,15 +3,17 @@
     <LinkTo @route={{this.router.currentRoute.localName}} data-test-secrets-tab="Secrets">Secrets</LinkTo>
     <LinkTo @route="configuration" data-test-secrets-tab="Configuration">Configuration</LinkTo>
   </:tabLinks>
-  {{! TODO checks for if list >0 show filter }}
+
   <:toolbarFilters>
     {{#unless @noMetadataListPermissions}}
-      <KvListFilter
-        @secrets={{@secrets}}
-        @mountPoint={{this.mountPoint}}
-        @filterValue={{@filterValue}}
-        @pageFilter={{@pageFilter}}
-      />
+      {{#if (or @secrets @filterValue)}}
```

Comment:
I need to have the `or` for `filterValue` so users can clear the filter if no secrets exist. 
![image](https://github.com/hashicorp/vault/assets/6618863/49b7cf61-96b9-41c2-918c-d303f40d7326)